crates/api_common/src/utils.rs

```diff
@@ -835,13 +840,101 @@ fn limit_expire_time(expires: DateTime<Utc>) -> LemmyResult<Option<DateTime<Utc>
   }
 }
 
+pub async fn process_markdown(
+  text: &str,
+  slur_regex: &Option<Regex>,
+  context: &LemmyContext,
+) -> LemmyResult<String> {
+  let text = remove_slurs(text, slur_regex);
+  let (text, links) = markdown_rewrite_image_links(text);
+  RemoteImage::create(&mut context.pool(), links).await?;
```

Comment:
RemoteImage is only written if the image is getting proxied. The table only exists to check that this url is valid for proxying, so there is no need to create it otherwise.